lib/portlayer/event/collector/vsphere/collector.go

```diff
@@ -102,10 +104,17 @@ func (ec *EventCollector) monitoredObjects() []types.ManagedObjectReference {
 	return refs
 }
 func (ec *EventCollector) Stop() {
-	_, err := ec.vmwManager.Destroy(context.Background())
-	if err != nil {
-		log.Warnf("%s failed to destroy the govmomi manager: %s", name, err.Error())
-	}
+	// End the event collection
+	ec.cancel()
+
+	// The EventManager cannot be destroyed like this as it isn't a ManagedEntity.
+	// TODO: we do need to ensure the EventHistoryCollector is destroyed, but that's a specific call
+	// and requires a govmomi change. At least it is lifecycle coupled with the session.
```

Comment:
Can we file an issue for this TODO and cross-link it with this code change?

Reply:
Fixed in PR vmware/govmomi#962 - comment just needs removing.